aar_builder/merge_aar.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python
+#
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Merge a list of srcaar files into one
+
+
+Example usage:
+  python merge_aar.py --inputs=<srcaar1> --inputs=<srcaar2> --output=<srcaar output>
+"""
+import os
+import stat
+import zipfile
+import tempfile
+import shutil
+import subprocess
+from absl import app, flags, logging
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_multi_string(
+    "inputs", None,
+    "A list of srcaars to merge together.")
+flags.DEFINE_string(
+    'output', None,
+    "Output file location for merged srcaar file."
+)
+
+
+def main(argv):
+  if len(argv) > 1:
+    raise app.UsageError('Too many command-line arguments.')
+  input_srcaars = FLAGS.inputs
+  if len(input_srcaars) <= 1:
+    raise app.UsageError(
+        'Input srcaars needs more than 1 entry, currently only %d.'.format(len(input_srcaars)))
+  base_temp_dir = tempfile.mkdtemp()
+
+  for input in input_srcaars:
```

Comment:
It'd be good to add comments to this, to explain what it is doing. Same with the re-zipping below.

Reply:
Done